src/Stratis.Bitcoin.Features.RPC/Controllers/RPCController.cs

```diff
@@ -90,27 +90,40 @@ private RPCResponse SendRPCRequest(RPCRequest request)
         /// <summary>
         /// Call an RPC method by name.
         /// </summary>
+        /// <param name="body">The request to process.</param>
         /// <returns>A JSON result that varies depending on the RPC method.</returns>
         [Route("callbyname")]
         [HttpPost]
         public IActionResult CallByName([FromBody]JObject body)
         {
+            Guard.NotNull(body, nameof(body));
+
             try
             {
-                var ignoreCase = StringComparison.InvariantCultureIgnoreCase;
+                StringComparison ignoreCase = StringComparison.InvariantCultureIgnoreCase;
                 string methodName = (string)body.GetValue("methodName", ignoreCase);
+
                 ControllerActionDescriptor actionDescriptor = null;
                 if (!this.GetActionDescriptors()?.TryGetValue(methodName, out actionDescriptor) ?? false)
                     throw new Exception($"RPC method '{ methodName }' not found.");
                 
                 // Prepare the named parameters that were passed via the query string in the order that they are expected by SendCommand.
                 List<ControllerParameterDescriptor> paramInfos = actionDescriptor.Parameters.OfType<ControllerParameterDescriptor>().ToList();
 
-                object[] paramsAsObjects = paramInfos.Select(paramInfo => 
-                    body.TryGetValue(paramInfo.Name, ignoreCase, out JToken jValue) && !jValue.HasValues
-                        ? jValue.ToString()
-                        : paramInfo.ParameterInfo.DefaultValue?.ToString())
-                    .Cast<object>().ToArray();
+                var paramsAsObjects = new object[paramInfos.Count];
```

Comment:
Bad taste creates many more millionaires than good taste.
*Charles Bukowski* 😂

Reply:
we're on then!